bhttp/src/http_compat/encode.rs

```diff
@@ -0,0 +1,441 @@
+                            // Is a trailers frame
+                            let mut buf = Vec::new();
+                            // First, we need to write zero-length chunk to indicate end, since we always using Mode::IndeterminateLength
+                            write_len(0, &mut buf)?;
+
+                            // Then, write the trailer field section with indeterminate length mode
+                            let field_section = FieldSection::try_from(trailers)?;
+                            field_section.write_bhttp(Mode::IndeterminateLength, &mut buf)?;
+
+                            // Switch to Done state since we have written the trailers
+                            this.state = HttpEncodeState::Done;
+                            Poll::Ready(Some(Ok(buf)))
+                        } else {
+                            // For unknown frame types, report an invalid data error.
+                            this.state = HttpEncodeState::Done;
+                            Poll::Ready(Some(Err(Error::UnknownHttpBodyFrameType)))
+                        }
+                    }
+                    Poll::Ready(Some(Err(e))) => Poll::Ready(Some(Err(Error::Io(
+                        std::io::Error::new(std::io::ErrorKind::Other, e),
+                    )))),
+                    Poll::Ready(None) => {
+                        // End of body, write zero-length chunk to indicate end
+                        let mut end_chunk = Vec::new();
+                        write_len(0, &mut end_chunk)?;
+                        this.state = HttpEncodeState::Done;
+                        Poll::Ready(Some(Ok(end_chunk)))
+                    }
+                    Poll::Pending => Poll::Pending,
+                }
+            }
+            HttpEncodeState::Done => Poll::Ready(None),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bytes::Bytes;
+    use futures::executor::block_on;
+    use futures::StreamExt;
+    use http::{Request, Response};
+    use http_body::Frame;
+    use http_body_util::Full;
+    use http_body_util::StreamBody;
+
+    #[test]
+    fn test_request_to_bhttp_conversion() {
+        let request = Request::builder()
+            .method("GET")
+            .uri("https://example.com/path")
+            .header("User-Agent", "test-agent")
+            .header("Accept", "application/json")
+            .body(Full::new(Bytes::from("test body")))
+            .unwrap();
+
+        let stream = BhttpEncoder::from_request(request);
+        let future = stream.fold(Vec::new(), |mut acc, item| async move {
+            if let Ok(data) = item {
+                acc.extend(data);
+            }
+            acc
+        });
+
+        let bhttp_data = block_on(future);
+
+        // Verify that we got some data
+        assert!(!bhttp_data.is_empty());
+
+        // The data should be longer than just the body since it includes headers and metadata
+        assert!(bhttp_data.len() > 9);
```

Comment:
This seems pretty weak in terms of its utility as a test.  I suggest that you have a set of known-answer tests for these.  On the one hand, `http::Request`, on the other a byte array.  Those can live in a common test module.  This code can run down the list and check that the output matches the expectation.